File: apps/android/app/src/main/java/com/barns/app/data/repository/MockCareRepository.kt
package com.barns.app.data.repository

import com.barns.app.domain.model.CareLog
import com.barns.app.domain.model.CareRepeatRule
import com.barns.app.domain.model.CareTask
import com.barns.app.domain.model.CareTaskStatus
import com.barns.app.domain.model.CareType
import com.barns.app.domain.repository.CareRepository
import java.time.Instant
import java.time.temporal.ChronoUnit

/**
 * In-memory, local-only care data. No persistence, no network.
 * Seed data is fake, non-sensitive, and image-null compatible.
 */
class MockCareRepository : CareRepository {
    private val lock = Any()
    private val tasks = mutableListOf(
        CareTask(
            id = "care-task-001",
            productItemId = "item-wall-green-001",
            title = "Inspect wall greenery",
            careType = CareType.INSPECTION,
            dueDate = Instant.now().plus(2, ChronoUnit.DAYS),
            repeatRule = CareRepeatRule.MONTHLY,
            status = CareTaskStatus.PENDING,
            reminderEnabled = false,
            imageUrl = null,
        ),
        CareTask(
            id = "care-task-002",
            productItemId = "item-interior-green-001",
            title = "Water desk planter",
            careType = CareType.WATERING,
            dueDate = Instant.now().plus(5, ChronoUnit.DAYS),
            repeatRule = CareRepeatRule.WEEKLY,
            status = CareTaskStatus.PENDING,
            reminderEnabled = false,
            imageUrl = null,
        ),
    )
    private val logs = listOf(
        CareLog(
            id = "care-log-001",
            productItemId = "item-wall-green-001",
            careTaskId = null,
            careType = CareType.CLEANING,
            performedAt = Instant.now().minus(7, ChronoUnit.DAYS),
            memo = null,
            imageUrl = null,
        ),
    )

    override suspend fun careTasks(): List<CareTask> =
        synchronized(lock) { tasks.toList() }

    override suspend fun careTask(id: String): CareTask? =
        synchronized(lock) { tasks.firstOrNull { it.id == id } }

    override suspend fun completeCareTask(id: String) {
        synchronized(lock) {
            val index = tasks.indexOfFirst { it.id == id }
            if (index >= 0) {
                tasks[index] = tasks[index].copy(status = CareTaskStatus.COMPLETED)
            }
        }
    }

    override suspend fun careLogs(productItemId: String?): List<CareLog> =
        synchronized(lock) {
            if (productItemId == null) logs else logs.filter { it.productItemId == productItemId }
        }
}

File: apps/ios/Barns/BarnsApp/Data/Repositories/MockCareRepository.swift
import Foundation

/// In-memory, local-only care data. No persistence, no network.
/// Seed data is fake, non-sensitive, and image-null compatible.
actor MockCareRepository: CareRepository {
    private var tasks: [CareTask]
    private let logs: [CareLog]

    init() {
        let now = Date()
        let day: TimeInterval = 60 * 60 * 24

        tasks = [
            CareTask(
                id: "care-task-001",
                productItemId: "item-wall-green-001",
                title: "Inspect the entryway green wall",
                careType: .inspection,
                dueDate: now.addingTimeInterval(2 * day),
                repeatRule: .monthly,
                status: .pending,
                reminderEnabled: false,
                imageUrl: nil
            ),
            CareTask(
                id: "care-task-002",
                productItemId: "item-interior-green-001",
                title: "Water the reception planter",
                careType: .watering,
                dueDate: now.addingTimeInterval(5 * day),
                repeatRule: .weekly,
                status: .pending,
                reminderEnabled: false,
                imageUrl: nil
            )
        ]

        logs = [
            CareLog(
                id: "care-log-001",
                productItemId: "item-wall-green-001",
                careTaskId: nil,
                careType: .cleaning,
                performedAt: now.addingTimeInterval(-7 * day),
                memo: "Wiped dust from the panel leaves.",
                imageUrl: nil
            )
        ]
    }

    func careTasks() async throws -> [CareTask] {
        tasks
    }

    func careTask(id: CareTask.ID) async throws -> CareTask? {
        tasks.first { $0.id == id }
    }

    func completeCareTask(id: CareTask.ID) async throws {
        guard let index = tasks.firstIndex(where: { $0.id == id }) else { return }
        let task = tasks[index]
        tasks[index] = CareTask(
            id: task.id,
            productItemId: task.productItemId,
            title: task.title,
            careType: task.careType,
            dueDate: task.dueDate,
            repeatRule: task.repeatRule,
            status: .completed,
            reminderEnabled: task.reminderEnabled,
            imageUrl: task.imageUrl
        )
    }

    func careLogs(for productItemId: ProductItem.ID?) async throws -> [CareLog] {
        guard let productItemId else { return logs }
        return logs.filter { $0.productItemId == productItemId }
    }
}
